pkg/cache/internal/informers_map_test.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+type testKind struct {
+	gvk schema.GroupVersionKind
+}
+
+func (o *testKind) SetGroupVersionKind(kind schema.GroupVersionKind) {
+	o.gvk = kind
+}
+
+func (o *testKind) GroupVersionKind() schema.GroupVersionKind {
+	return o.gvk
+}
+
+type testType struct {
+	kind  schema.ObjectKind
+	label string
+}
+
+func newTestType(label string) testType {
+	return testType{
+		kind: &testKind{
+			gvk: schema.GroupVersionKind{
+				Group:   "originalgroup",
+				Version: "OriginalVersion",
+				Kind:    "OriginalKind",
+			},
+		},
+		label: label,
+	}
+}
+
+func (obj testType) GetObjectKind() schema.ObjectKind { return obj.kind }
+func (obj testType) DeepCopyObject() runtime.Object   { return obj }
+
+// TestGVKFixupWatcher tests that gvkFixupWatcher behaves like the watch that it
+// wraps and that it overrides the GVK.
+// Adapted from https://github.com/kubernetes/kubernetes/blob/adbda068c1808fcc8a64a94269e0766b5c46ec41/staging/src/k8s.io/apimachinery/pkg/watch/watch_test.go#L33-L78
+func TestGVKFixupWatcher(t *testing.T) {
```

Comment:
AFAIK, the rest of controller-runtime's tests use Ginkgo/Gomega. Would you mind refactoring this test to be consistent with the rest of the repo?

Reply:
I've re-implemented the tests using Gomega `Expect` , but tried to retain the spirit of the original tests so that they can be easily compared to the original.

Reply:
Also replaced the convoluted fake type with actual `metav1.PartialObjectMetadata`